### PR TITLE
Add api for Dedupe.getduplicates

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
Exposes duplicate retrieval to the api

Before
----------------------------------------
Not api-gettable

After
----------------------------------------
Api Dedupe.getduplicates works

Technical Details
----------------------------------------

Comments
----------------------------------------

### CODE REVIEW

File: api/v3/Dedupe.php [hunk 1]
```diff
@@ -137,3 +137,51 @@ function _civicrm_api3_dedupe_getstatistics_spec(&$params) {
...
+ * @param array $params
+ *   Array of parameters determined by getfields.
+ */
+function _civicrm_api3_dedupe_getduplicates_spec(&$params) {
+  $params['rule_group_id'] = [
+    'title' => ts('Rule Group ID'),
+    'api.required' => TRUE,
+    'type' => CRM_Utils_Type::T_INT,
+  ];
+  $params['group_id'] = [
```
Comment (pfigel) [inline api/v3/Dedupe.php]: Parameter is defined here, but not used as a filter in `civicrm_api3_dedupe_getduplicates`.

**Review by pfigel — CHANGES_REQUESTED** [review]
Aside from the `group_id` issue, this LGTM.

(*CiviCRM Review Template WORD-1.2*)

* General standards
    * ([`r-explain`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-explain)) __PASS__
    * ([`r-user`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-user)) __PASS__: No user impact, new API
    * ([`r-doc`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-doc)) __PASS__
    * ([`r-run`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-run)) __ISSUE__: group_id parameter is defined but not used to filter results. `r-run` successful otherwise, results match existing UI.
* Developer standards
    * ([`r-tech`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-tech)) __PASS__: Doesn't change any behaviour. Allows decoupling / use of feature in extension.
    * ([`r-code`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-code)) __PASS__
    * ([`r-maint`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-maint)) __PASS__
    * ([`r-test`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-test)) __PASS__

**Comment by eileenmcnaughton** [comment]
Thanks @pfigel - I removed the 'group' parameter & enhanced the 'criteria' description to point out group can be included in the criteria (which I think is better since I don't think the underlying code really does need 2 filter methodologies)

**Comment by eileenmcnaughton** [comment]
test this please

**Comment by eileenmcnaughton** [comment]
test this please

**Comment by eileenmcnaughton** [comment]
unrelated fail

**Pull request merged**